### PR TITLE
Update Airflow Release Doc

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Minor update of the airflow release doc

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

File: dev/README_RELEASE_AIRFLOW.md [hunk 1]
```diff
@@ -199,11 +199,14 @@ The Release Candidate artifacts we vote upon should be the exact ones we vote ag
 
 - Check out the 'test' branch
 
+  For major/minor version release, please follow  the instructions at [[\Optional\] Prepare new release branches and cache](#%5Coptional%5C-prepare-new-release-branches-and-cache) to create the 'test' and 'stable' branches.
```
Comment (kaxil) [inline dev/README_RELEASE_AIRFLOW.md]:
```suggestion
  For major/minor version release, please follow  the instructions at [Prepare new release branches and cache](#%5Coptional%5C-prepare-new-release-branches-and-cache) to create the 'test' and 'stable' branches.
```

Comment (kaxil) [reply dev/README_RELEASE_AIRFLOW.md]: We don't need optional in anchor text

File: dev/README_RELEASE_AIRFLOW.md [hunk 2]
```diff
@@ -199,11 +199,14 @@ The Release Candidate artifacts we vote upon should be the exact ones we vote ag
...
+
     ```shell script
     git checkout v${VERSION_BRANCH}-test
     ```
 
-- Set your version to 2.0.N in `setup.py` (without the RC tag)
+- Set your version in `setup.py` (without the RC tag)
+- Modify `./scripts/ci/pre_commit/pre_commit_supported_versions.py` and let pre-commit do the job
```
Comment (kaxil) [inline dev/README_RELEASE_AIRFLOW.md]:
```suggestion
- Add supported Airflow version to `./scripts/ci/pre_commit/pre_commit_supported_versions.py` and let pre-commit do the job
```

**Review by kaxil — APPROVED** [review]
Minor comments suggested, feel free to merge once those are addresed

File: dev/README_RELEASE_AIRFLOW.md [hunk 3]
```diff
@@ -584,7 +588,7 @@ Hey fellow Airflowers,
 
 I have cut Airflow ${VERSION}. This email is calling a vote on the release,
 which will last for 72 hours, from Friday, October 8, 2021 at 4:00 pm UTC
```
Comment (jedcunningham) [inline dev/README_RELEASE_AIRFLOW.md]:
```suggestion
which will last at least 72 hours, from Friday, October 8, 2021 at 4:00 pm UTC
```

**Review by jedcunningham — APPROVED** [review]

**Pull request merged**